state.go

```diff
@@ -422,19 +461,66 @@ type AuthorizedState struct {
 	BaseState
 }
 
-func (a *AuthorizedState) Handle(c Controller) (State, bool) {
-	// TODO HasUpgrade should return update information
-	has, err := c.HasUpgrade()
+func (a *AuthorizedState) Handle(ctx *StateContext, c Controller) (State, bool) {
+	// restore previous state information
+	sd, err := LoadStateData(ctx.store)
+
+	// tricky part - try to figure out if there's an update in progress, if so
+	// proceed to UpdateCommitState; in case of errors that occur either now or
+	// when the update was being feched/installed previously, try to handle them
+	// gracefully
+
+	// handle easy case first, no update info present, means no update in progress
+	if err != nil && os.IsNotExist(err) {
+		log.Debug("no update in progress, proceed")
+		return updateCheckWaitState, false
+	}
+
 	if err != nil {
-		log.Errorf("has upgrade check failed: %s", err)
-		// we may or may now have an upddate ready
-		return NewErrorState(err), false
+		log.Errorf("failed to restore update information: %v", err)
+		me := NewFatalError(errors.Wrapf(err, "failed to restore update information"))
+
+		// report update error with unknown deployment ID
+		// TODO: fill current image ID?
```

Comment:
Not really. This might be misleading. Better report that we don't know which id failed. Maybe server will be able to figure out what was last update id.